addons/test_mail/tests/test_mail_tools.py

```diff
@@ -81,3 +82,129 @@ def test_find_partner_from_emails(self):
+        sources = [
+            (self._test_email, True),  # normalized
+            (self._test_email, False),  # normalized
+            (encapsulated_test_email, True),  # encapsulated, same
+            (encapsulated_test_email, False),  # encapsulated, same
+            (f'"AnotherName" <{self._test_email}', True),  # same normalized, other name
+            (f'"AnotherName" <{self._test_email}', False),  # same normalized, other name
+        ]
+        expected = [follower_partner, test_partner,
+                    follower_partner, test_partner,
+                    follower_partner, test_partner,
+                    follower_partner, test_partner]
+        for (source, follower_check), expected in zip(sources, expected):
+            with self.subTest(source=source, follower_check=follower_check):
+                # test direct match
+                partner = self.env['res.partner']._mail_find_partner_from_emails(
+                    [source], records=linked_record if follower_check else None
+                )[0]
+                self.assertEqual(partner, expected,
+                                'Mail: formatted email is recognized through usage of normalized email')
+
+        # multi-email
+        _test_email_2 = '"Robert Astaire" <not.alfredoastaire@test.example.com>'
+        (follower_partner + test_partner).sudo().write({'email': f'{self._test_email}, {_test_email_2}'})
+        sources = [
+            (self._test_email, True),  # first email
+            (self._test_email, False),  # first email
+            (_test_email_2, True),  # second email
+            (_test_email_2, False),  # second email
+            ('not.alfredoastaire@test.example.com', True),  # normalized second email in field
+            ('not.alfredoastaire@test.example.com', False),  # normalized second email in field
+            (f'{self._test_email}, {_test_email_2}', True),  # direct match
+            (f'{self._test_email}, {_test_email_2}', False)  # direct match
+        ]
+        expected = [self.env['res.partner'], self.env['res.partner'],
+                    self.env['res.partner'], self.env['res.partner'],
+                    self.env['res.partner'], self.env['res.partner'],
+                    self.env['res.partner'], self.env['res.partner']]
+        for (source, follower_check), expected in zip(sources, expected):
+            with self.subTest(source=source, follower_check=follower_check):
+                # test direct match
+                partner = self.env['res.partner']._mail_find_partner_from_emails(
+                    [source], records=linked_record if follower_check else None
+                )[0]
+                self.assertEqual(partner, expected,
+                                'Mail (FIXME): not recognized due to usage of email_normalize that does not accept multi emails')
+
+    @users('employee')
+    def test_find_partner_from_emails_users_multiemail(self):
```

Comment:
Not sure why this one has "multiemail" in the name and not the previous one, they both seem to have the same kind of testing logic.